### PR TITLE
docs: troubleshooting section revamp (close #4052)

### LINKED ISSUE #4052: Troubleshooting section revamp
Remove the current troubleshooting guide because it's not useful and far from complete.

Replace it with a generic debugging guide:
- Check appropriate logs (links to server logs, console logs, remote service logs etc.)
- Check Github issues
- Check Hasura blog
- Check Postgres docs
- Jump on Discord for help

Create a useful troubleshooting guide including errors: https://github.com/hasura/graphql-engine/issues/2871

### PR DESCRIPTION
**Pull request by marionschleifer**
### Description

This PR removes the incomplete troubleshooting guide from the docs.

It adds a temporary page that contains references as to where to find solutions on errors.

Later we'll replace this with a comprehensive page with all errors.

### Changelog

- [x] `CHANGELOG.md` is updated with user-facing content relevant to this PR.

### Affected components

- [x] Docs

### Related Issues

https://github.com/hasura/graphql-engine/issues/4052

### Steps to test and verify

https://deploy-preview-4825--hasura-docs.netlify.app/graphql/manual/troubleshooting/index.html

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -97,6 +97,7 @@ using this flag is insecure since verification is not carried out.
 - cli: remove irrelevant flags from init command (close #4508) (#4549)
 - docs: update migrations docs with config v2 (#4586)
 - docs: update actions docs (#4586)
+- docs: replace incomplete trouble shooting section with reference on debugging errors (close #4052) (#4825)
```
Comment (tirumaraiselvan) [inline CHANGELOG.md:100]: This appears to be in wrong place.

File: docs/graphql/manual/troubleshooting/index.rst [hunk 2]
```diff
@@ -12,118 +12,58 @@ Troubleshooting Hasura GraphQL engine errors
...
-the ``author`` table, the Hasura GraphQL engine will create a special field ``author`` in the ``article`` table in its
-internal state. This field will be available via the GraphQL interface.
+Discord
+-------
 
-When this table is described using ``psql``, the ``author`` field will not be available as part of the list of fields
-returned by the describe command as it is something added by Hasura GraphQL engine. Now if a new column is created
-with the same name, i.e. ``author``, via ``psql``, the Hasura GraphQL engine will throw the above error when restarted as it has two
-references to the ``author`` field for the ``article`` table.
+If you didn't find a solution in any of the abovementioned sections or if you prefer to troubleshoot with the community,
```
Comment (rikinsk) [inline docs/graphql/manual/troubleshooting/index.rst:66]: Typo in: "above mentioned"

Comment (marionschleifer) [reply docs/graphql/manual/troubleshooting/index.rst]: It's actually a word: https://www.merriam-webster.com/dictionary/abovementioned :D but it's fine for me to change :)

Comment (rikinsk) [reply docs/graphql/manual/troubleshooting/index.rst]: Wow. TIL. In that case sorry for changing it :)

**Pull request merged**